### PR TITLE
Rename Protobuf_PROTOC_EXECUTABLE to PROTOBUF_PROTOC_EXECUTABLE for C…

### PR DESCRIPTION
**Pull request by JanuszL**
…make 3.5 compatibility

Update docker to use cmake 3.5

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

File: cmake/ProtobufGenCpp.cmake [hunk 1]
```diff
@@ -84,9 +84,9 @@ function(DALI_PROTOBUF_GENERATE_CPP SRCS HDRS)
     add_custom_command(
       OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}.pb.cc"
              "${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}.pb.h"
-      COMMAND  ${Protobuf_PROTOC_EXECUTABLE}
+      COMMAND  ${PROTOBUF_PROTOC_EXECUTABLE}
```
Comment (gzygzy9211) [inline cmake/ProtobufGenCpp.cmake]: In fact, only in cmake 3.5, ```${PROTOBUF_PROTOC_EXECUTABLE}``` is used while in 3.6 OR HIGHER, ```${Protobuf_PROTOC_EXECUTABLE}``` is used, so I would rather add 
```
set(Protobuf_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE})
```
at the beginning of this function if the CMake version is 3.5

Comment (cliffwoolley) [reply cmake/ProtobufGenCpp.cmake]: That seems reasonable.  (In as much as following this breaking change they did in CMake is "reasonable".  :) )

Comment (gzygzy9211) [reply cmake/ProtobufGenCpp.cmake]: Package-specific functions of CMake seems not to be so stable as we think. T^T

Comment (cliffwoolley) [reply cmake/ProtobufGenCpp.cmake]: Hmm -- actually it looks like they defined the all-upper-case version as well when renaming for backward compatibility.  I'm attempting to prove whether all versions after 3.5 had that.

**Pull request merged**